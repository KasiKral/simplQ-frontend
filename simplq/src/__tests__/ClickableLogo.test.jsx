import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Logo from '../components/common/ClickableLogo';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(<Logo />);
  expect(asFragment()).toMatchSnapshot();
});

it('click event', () => {
  const click = jest.fn();
  const { getByText, container } = render(<Logo onClick={click} />);
  fireEvent.click(getByText('SimplQ'));
  fireEvent.click(container.firstChild);
  fireEvent.click(container.querySelector('img'));
  // User clicked on 3 different dom elements
  expect(click).toHaveBeenCalledTimes(3);
});

it("component's elements are setup correctly", () => {
  const { container } = render(<Logo />);
  expect(container.querySelector('img')).toHaveAttribute('src');
  expect(container.querySelector('img')).toHaveAttribute('alt');
  expect(container.firstChild).toHaveClass('logo');
});
