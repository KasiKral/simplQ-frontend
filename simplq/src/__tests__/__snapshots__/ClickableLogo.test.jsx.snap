// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="logo"
  >
    <img
      alt="Home"
      src="/images/Simple-Q.png"
    />
    <p>
      SimplQ
    </p>
  </div>
</DocumentFragment>
`;
